parser/parser.y

```diff
@@ -5421,9 +5421,18 @@ Type:
 NumericType:
 	IntegerType OptFieldLen FieldOpts
 	{
+		// For boolean and bool, mysql maps them to tinyint(1)
+		// In tidb, tinyint has 4 bits by default. This if branch is to
+		// handle this situation.
+		var x *types.FieldType
+		if b, ok :=  $1.(*types.FieldType); ok {
+			x = b
+			x.Flen = 1
```

Comment:
The `Flen` should be set in `Bool` or `Boolean` entry.

Reply:
@coocood PTAL again. I just change to a new to fix this bug.

Reply:
And the return type of an item should be consistent.